pkg/build/api/v1beta1/types.go

```diff
@@ -223,6 +228,17 @@ type WebHookTrigger struct {
 	Secret string `json:"secret,omitempty" yaml:"secret,omitempty"`
 }
 
+// BuildTriggerImageChangeParams represents the parameters to the ImageChange trigger.
+type ImageChangeTrigger struct {
+	// Image is used to specify the value in the BuildConfig to replace with the 
+	// immutable image id supplied by the ImageRepository when this trigger fires.
+	Image string `json:"image,omitempty" yaml:"image,omitempty"`
+	// ImageRepositoryRef a reference to a Docker image repository to watch for changes.
+	ImageRepositoryRef kapi.ObjectReference `json:"imageRepositoryRef,omitempty" yaml:"imageRepositoryRef,omitempty"`
+	// Tag is the name of an image repository tag to watch for changes.
```

Comment:
Right, if deployment is defaulting "" to "latest" we should too.  I think there's value in not requiring people to specify it, and some cost for clients to have to deal with it.  If you don't default, everyone will pick their own (Git i.e. uses branch master by default, and origin as the remote by default, so everyone deals with it).

----- Original Message -----
> > @@ -223,6 +228,17 @@ type WebHookTrigger struct {
> >  	Secret string `json:"secret,omitempty" yaml:"secret,omitempty"`
> >  }
> >  
> > +// BuildTriggerImageChangeParams represents the parameters to the
> > ImageChange trigger.
> > +type ImageChangeTrigger struct {
> > +	// Image is used to specify the value in the BuildConfig to replace with
> > the
> > +	// immutable image id supplied by the ImageRepository when this trigger
> > fires.
> > +	Image string `json:"image,omitempty" yaml:"image,omitempty"`
> > +	// ImageRepositoryRef a reference to a Docker image repository to watch
> > for changes.
> > +	ImageRepositoryRef kapi.ObjectReference
> > `json:"imageRepositoryRef,omitempty" yaml:"imageRepositoryRef,omitempty"`
> > +	// Tag is the name of an image repository tag to watch for changes.
> 
> I haven't looked at all the other use cases, but in this use case, the only
> way we can make it non-optional is if we default to "latest".  i'm fine with
> doing that if you prefer it, but there's no way to perform image change
> trigger logic without some tag value to be watching for changes.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/557/files#r22607360